dialoptions.go

```diff
@@ -246,21 +247,34 @@ func WithServiceConfig(c <-chan ServiceConfig) DialOption {
 	})
 }
 
+// WithConnectParams configures the dialer to use the provided ConnectParams.
```

Comment:
Is this the best place to document our defaults?

Reply:
Made the comments a bit more verbose.